dfastmi/io/RiverParameterData.py

```diff
@@ -219,183 +226,175 @@ def collect_values1(
 
         Returns
         -------
-        data : List[List[float]]
+        data : int
             A list of lists. Each list contains per reach within the corresponding
             branch one float.
         """
         self.is_initialized()
-        try:
-            g_val = self._config["General"][key]
-        except:
-            g_val = ""
-
-        all_values = []
-        for ib in range(len(self._branches)):
-            branch = self._branches[ib]
+        val = self.__read_value(key, branch, reach)
+        if val == "" and default is not None:
+            ival = default
+        else:
             try:
-                b_val = self._config[branch][key]
+                vals = tuple(int(x) for x in val.split())
```

Comment:
feels similar to read_key_bool

Reply:
Discussed, we will leave these comments for a later refactoring.
This will be tackled in https://issuetracker.deltares.nl/browse/DFAST-112